### DISCUSSION TITLE
Consider removing "Async" suffixed from generated commands

### DISCUSSION THREAD

**Discussion by taori** [CommunityToolkit/dotnet#204] [General]
Naming methods "METHODNAMEAsync" is a common naming convention for methods. I think the code generator should not name the commands with the async suffix too, since that is just sort of redundant

**Comment by Sergio0694** [comment]
But... It already does 🤔

Do you have a minimal repro where that is not working for you?

**Comment by taori** [comment]
![grafik](https://user-images.githubusercontent.com/5545184/162281812-50a0fba2-457c-4034-b543-cb891a963de3.png)

8.0.0-Build.138

**Reply by Sergio0694** [reply]
Well, as you can see that method is _not_, in fact, asynchronous. If you change it to return a `Task`, you'll see the name will be handled properly, and the generated command will be an `IAsyncRelayCommand` instance 🙂

**Reply by taori** [reply]
My bad in the sample - for "public async void" it is not removed, for "public async Task" it works :)

**Reply by Sergio0694** [reply]
I would say that's by design to push people to use `async Task` and not `async void` for commands (and in general) 😄

**Reply by taori** [reply]
Hm, while i am inclined to insist that is a bug, i agree too much about discouraging people from using async void, because in most cases where people are not aware of the implications of async void it just introduces hard to find bugs

**Reply by Sergio0694** [reply]
> "while i am inclined to insist that is a bug"

I mean it's not a bug, we're explicitly only checking this if the method returns a `Task` 😄

https://github.com/CommunityToolkit/dotnet/blob/f3bd3d8b72615f1c1a5cd02876518ef01de518e5/CommunityToolkit.Mvvm.SourceGenerators/Input/ICommandGenerator.Execute.cs#L352-L357

**Reply by taori** [reply]
A bug in terms of naming guidelines, not a code bug in terms of behavior, yeah. Definitely true. I'll drop it since i agree on your motived either way :)

**Reply by Sergio0694** [reply]
I mean you do make a good point, it would be nice to still give users some better info here. We can't do this for 8.0.0 since we're feature-locked for that now, but I'm thinking we might look into adding some more analyzers with some warnings or whatnot for post 8.0.0. For instance, here we could show a warning suggesting to change the method to return a `Task`, instead of just ignoring it.
